lib/Storage.js

```diff
@@ -1,87 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
 const {Cu} = require('chrome');
-var {indexedDB} = require('sdk/indexed-db');
+Cu.import('resource://gre/modules/XPCOMUtils.jsm'); /* global XPCOMUtils */
 
-exports.Storage = function(prefix, sandbox=null) {
-  if (sandbox === null) {
-    sandbox = {Promise};
-  }
+XPCOMUtils.defineLazyModuleGetter(this, 'JSONFile',
+                                  'resource://gre/modules/JSONFile.jsm'); /* global JSONFile */
 
-  const idbPromise = new sandbox.Promise((resolve, reject) => {
-    let request = indexedDB.open('normandy', 1);
-    request.onsuccess = event => resolve(event.target.result);
-    request.onerror = err => reject(err);
+XPCOMUtils.defineLazyModuleGetter(this, 'OS',
+                                  'resource://gre/modules/osfile.jsm'); /* global OS */
+XPCOMUtils.defineLazyModuleGetter(this, 'Task',
+                                  'resource://gre/modules/Task.jsm'); /* global Task */
 
-    request.onupgradeneeded = event => {
-      var db = event.target.result;
-      db.createObjectStore('storage', {keyPath: 'key'});
-    };
-  });
+const {Log} = require('./Log.js');
+
+let storePromise;
+
+function loadStorage() {
+  if (storePromise === undefined) {
+    let path = OS.Path.join(OS.Constants.Path.profileDir, 'recipe-client.json');
```

Comment:
There aren't really very good existing practices that I'm aware of, beyond "if you need more than 1-2 files, use a folder, otherwise, put them directly in the profile directory."

Ideally it should be easy to infer from the filename what's in it. Good examples: cookies.sqlite, signons.db. Bad examples: XUL.mfasl (or whatever extension that had - the fastload file is stupid).

I would suggest that if the project's official public name (rather than codebase codename like "normandy") is "SHIELD", then that should probably be in the name of the file (probably all-lowercase to avoid SHOUTING :-) ) so that it is reasonably obvious to people who go and look at these files what their purpose is.